packages/relay/tests/lib/poller.spec.ts

```diff
@@ -0,0 +1,141 @@
+            ethImplStub.getLogs.returns(logs);
+            const loggerSpy = sandbox.spy(logger, 'info');
+
+            poller.add(tag, notifySubscriber);
+            done();
+            
+            expect(poller.isPolling()).to.be.true;
+            expect(poller.hasPoll(tag)).to.be.true;
+            expect(loggerSpy.calledTwice).to.be.true;
+            expect(loggerSpy.getCall(0).args[0]).to.equal('Poller: Polling for {"event":"logs","filters":{"address":"0x23f5e49569A835d7bf9AefD30e4f60CdD570f225","topics":["0xc8b501cbd8e69c98c535894661d25839eb035b096adfde2bba416f04cc7ce987"]}}');
+            expect(loggerSpy.getCall(1).args[0]).to.equal('Poller: Starting polling');
+           
+        });
+
+        it('should stop polling', (done) => {
+            const loggerSpy = sandbox.spy(logger, 'info');            
+            poller.remove(tag);
+            done();
+
+            expect(poller.isPolling()).to.be.false;
+            expect(loggerSpy.callCount).to.equal(5);
+            expect(loggerSpy.getCall(0).args[0]).to.equal('Poller: No longer polling for {"event":"logs","filters":{"address":"0x23f5e49569A835d7bf9AefD30e4f60CdD570f225","topics":["0xc8b501cbd8e69c98c535894661d25839eb035b096adfde2bba416f04cc7ce987"]}}');
+            expect(loggerSpy.getCall(1).args[0]).to.equal('Poller: No active polls.');
+            expect(loggerSpy.getCall(2).args[0]).to.equal('Poller: Stopping polling');    
+            
+        });
+
+        it('should poll single line of log data', async () => {
+            const notifySubscriber = (tag, logs) =>{
+                logger.info(SINGLE_LINE);
+                return;
+            };
+
+            ethImplStub.getLogs.returns(logs);
+        
+            poller.add(tag, notifySubscriber);   
+            const loggerSpy = sandbox.spy(logger, 'info');
+            const poll = async() => {
+                poller.poll();
+            
+                expect(loggerSpy.callCount).to.equal(1);
+                expect(loggerSpy.getCall(0).args[0]).to.equal(FETCHING_DATA);
+            };
+
+            await poll();    
+            expect(loggerSpy.getCall(1).args[0]).to.equal(SINGLE_LINE);
+        });
+
+
+        it('should poll an array of log data', async () => {
+
+            const notifySubscriber = (tag, logsArray) =>{
+                logger.info(ARRAY_OF_LOGS);
+                return;
+            };
+
+            ethImplStub.getLogs.returns(logsArray);
+        
+            poller.add(tag, notifySubscriber);   
+            const loggerSpy = sandbox.spy(logger, 'info');
+            const poll = async() => {
+                poller.poll();
+            
+                expect(loggerSpy.callCount).to.equal(1);
+                expect(loggerSpy.getCall(0).args[0]).to.equal(FETCHING_DATA);                
+            };
+
+            await poll();   
+            expect(loggerSpy.getCall(1).args[0]).to.equal(ARRAY_OF_LOGS);
+        });
+
+    });
+
+});
```

Comment:
nit:  new line

Reply:
Done.

Reply:
Yeah there should be a line 135 with nothing on it.
Just makes it easier to git diff in the future.
It's a nit